app/src/main/java/org/schabi/newpipe/ktx/View.kt

```diff
@@ -332,6 +274,19 @@ fun View.animateHideRecyclerViewAllowingScrolling() {
     animate().alpha(0.0f).setDuration(200).start()
 }
 
+private class RunnableExecutorAdapter(
+    private val view: View,
+    private val execOnEnd: Runnable?,
+    private val shouldHideView: Boolean
+) : AnimatorListenerAdapter() {
+    override fun onAnimationEnd(animation: Animator) {
+        if (shouldHideView) {
+            view.isGone = true
+        }
+        execOnEnd?.run()
+    }
+}
```

Comment:
I'd create two classes, one for when `shouldHideView` is `true`, one for when it's `false`, in order to avoid that runtime check (maybe the second can extend the first, but idk). The names for the two classes should be imo: `HideAndExecOnEndListener` and `ExecOnEndListener`. `RunnableExecutorAdapter` is not so meaningful.